src/update_eh.cpp

```diff
@@ -187,12 +187,19 @@ bool fields_chunk::update_eh(field_type ft, bool skip_w_components) {
                  sizeof(realnum) * gv.ntot());
         }
 
-        if (f[ec][cmp] != f[dc][cmp])
+        if (f[ec][cmp] != f[dc][cmp]) {
           STEP_UPDATE_EDHB(f[ec][cmp], ec, gv, gvs_eh[ft][i].little_owned_corner0(ec),
                            gvs_eh[ft][i].big_corner(), dmp[dc][cmp], dmp[dc_1][cmp], dmp[dc_2][cmp],
                            s->chi1inv[ec][d_ec], dmp[dc_1][cmp] ? s->chi1inv[ec][d_1] : NULL,
                            dmp[dc_2][cmp] ? s->chi1inv[ec][d_2] : NULL, s_ec, s_1, s_2, s->chi2[ec],
                            s->chi3[ec], f_w[ec][cmp], dsigw, s->sig[dsigw], s->kap[dsigw]);
+
+          // special update for r=0 in cylindrical coordinates
+          STEP_UPDATE_EDHB0(f[ec][cmp], ec, gv, gvs_eh[ft][i].little_owned_corner(ec),
+                            gvs_eh[ft][i].big_corner(), dmp[dc][cmp], s->chi1inv[ec][d_ec], s_ec,
+                            s->chi2[ec], s->chi3[ec], f_w[ec][cmp], dsigw, s->sig[dsigw],
+                            s->kap[dsigw]);
```

Comment:
It occurs to me that we don't need a special `STEP_UPDATE_EDHB0` routine at all.  We just need to call `STEP_UPDATE_EDHB` and pass `NULL` for the off-diagonal terms.
```suggestion
          if (gv.dim == Dcyl) {
            vec is(gvs_eh[ft][i].little_owned_corner(ec));
            if (is.r() == 0) {
                vec ie(gvs_eh[ft][i].big_corner());
                ie.set_direction(R, 0);
                /* pass NULL for off-diagonal terms since they must be
                   zero at r=0 for an axisymmetric structure: */
                STEP_UPDATE_EDHB(f[ec][cmp], ec, gv, gvs_eh[ft][i].little_owned_corner0(ec),
                           gvs_eh[ft][i].big_corner(), dmp[dc][cmp], NULL, NULL,,
                           s->chi1inv[ec][d_ec], NULL, NULL, s_ec, s_1, s_2, s->chi2[ec],
                           s->chi3[ec], f_w[ec][cmp], dsigw, s->sig[dsigw], s->kap[dsigw]);
            }
          }
        }
```